xmtp_mls/src/groups/device_sync.rs

```diff
@@ -101,57 +107,145 @@ pub enum DeviceSyncError {
     UnspecifiedDeviceSyncKind,
     #[error("sync reply is too old")]
     SyncReplyTimestamp,
+    #[error(transparent)]
+    Subscribe(#[from] SubscribeError),
+}
+
+impl<ApiClient, V> Client<ApiClient, V>
+where
+    ApiClient: XmtpApi + 'static,
+    V: SmartContractSignatureVerifier + 'static,
+{
+    pub async fn start_sync_worker(
+        &self,
+        provider: &XmtpOpenMlsProvider,
+    ) -> Result<(), DeviceSyncError> {
+        self.sync_init(provider).await?;
+
+        crate::spawn(None, {
+            let client = self.clone();
+
+            let receiver = client.local_events.subscribe();
+            let mut sync_stream = Box::pin(receiver.stream_sync_messages());
+
+            async move {
+                while let Err(err) = client.sync_worker(&mut sync_stream).await {
+                    tracing::error!("Sync worker error: {err}");
```

Comment:
Question: What would happen if something were to happen here?

Reply:
It logs it and marches on to the next sync request. We don't want the sync worker to outright crash.